sql/parser/eval.go

```diff
@@ -70,17 +70,20 @@ func (d DBool) Type() string {
 
 // Compare implements the Datum interface.
 func (d DBool) Compare(other Datum) int {
+	if other == DNull {
+		// NULL compares lesser than any non-NULL value.
+		return 1
+	}
 	v, ok := other.(DBool)
 	if !ok {
-		// Anything other than a DBool compares greater (e.g. `true > NULL`).
-		return 1
+		panic(fmt.Sprintf("unsupported comparison: %s to %s", d.Type(), v.Type()))
```

Comment:
I'm not sure we can guarantee that an output column always has the same type or NULL. Consider:

```
SELECT CASE k WHEN 1 THEN 1 ELSE 'not one' END FROM t; 
```

Hmm, that doesn't actually work in postgres as postgres requires the value types to all match. We're not currently enforcing that. We probably should.